doc: Add note about secure faults missed by memfault if using TF-M (#7843)
nrfconnect/sdk-nrf: doc/nrf/libraries/others/memfault_ncs.rst

To get access to all the benefits, like up to 100 free devices connected, register at the `Memfault registration page`_.

.. note::

>    When building applications with  `Trusted Firmware M`_ (TF-M), the faults resulting from memory access in secure regions will not be caught by memfault's fault handler.

Reviewer: I think we either should say how to use SPM instead of TF-M or link to a doc page that explains this.

>    When building applications with  `Trusted Firmware M`_ (TF-M), the faults resulting from memory access in secure regions will not be caught by memfault's fault handler.
>    Instead they will be handled TF-M. This means that those faults wont be reported to the Memfault platform.

Reviewer 2: ```suggestion
   When building applications with :ref:`Trusted Firmware-M <ug_tfm>` (TF-M), the faults resulting from memory access in secure regions are not caught by Memfault's fault handler.
   Instead, they are handled by TF-M.
   This means that those faults are not reported to the Memfault platform.
```
